cmd/autoconfigbrancher/main.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/experiment/autobumper/bumper"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+)
+
+const (
+	githubOrg   = "openshift"
+	githubRepo  = "release"
+	githubLogin = "openshift-bot"
+	githubTeam  = "openshift/openshift-team-developer-productivity-test-platform"
+)
+
+var (
+	count = 0
+)
+
+type options struct {
+	githubLogin    string
+	githubToken    string
+	gitName        string
+	gitEmail       string
+	targetDir      string
+	assign         string
+	currentRelease string
+	futureReleases flagutil.Strings
+	debugMode      bool
+}
+
+func parseOptions() options {
+	var o options
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flag.StringVar(&o.githubLogin, "github-login", githubLogin, "The GitHub username to use.")
+	flag.StringVar(&o.githubToken, "github-token", "", "The path to the GitHub token file.")
+	flag.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	flag.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	flag.StringVar(&o.targetDir, "target-dir", "", "The directory containing the target repo.")
+	flag.StringVar(&o.assign, "assign", githubTeam, "The github username or group name to assign the created pull request to.")
+	flag.StringVar(&o.currentRelease, "current-release", "", "Configurations targeting this release will get branched.")
+	flag.Var(&o.futureReleases, "future-release", "Configurations will get branched to target this release, provide one or more times.")
+	flag.BoolVar(&o.debugMode, "debug-mode", false, "Enable the DEBUG level of logs if true.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login cannot be empty string")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	if o.targetDir == "" {
+		return fmt.Errorf("--target-dir is mandatory")
+	}
+	if o.assign == "" {
+		return fmt.Errorf("--assign is mandatory")
+	}
+	if o.currentRelease == "" {
+		return fmt.Errorf("--current-release is mandatory")
+	}
+	if len(o.futureReleases.Strings()) == 0 {
+		return fmt.Errorf("--future-release is mandatory")
+	}
+	return nil
+}
+
+func hasChanges() (bool, error) {
+	cmd := "git"
+	args := []string{"status", "--porcelain"}
+	logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
+	combinedOutput, err := exec.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		return false, err
```

Comment:
nit: if the command ever fails, I want to see its error output - please at least log it as debug like you do in `run()`

Reply:
I missed that one.